app/models/reports/pit/fy2018/base.rb

```diff
@@ -46,6 +46,26 @@ def describe_filter_as_html(_)
       ''
     end
 
+    def allowed_options(result)
+      options_from_result(result).keys.map(&:to_sym)
+    end
+
+    def filter_from_result(result)
+      # Old PITs don't the filter, but the HudFilterBase will handle display.
+      f = ::Filters::HudFilterBase.new(user_id: result.user_id)
+      f.update(options_from_result(result).with_indifferent_access)
+      f
+    end
+
+    private def options_from_result(result)
+      options = result.options.deep_dup
+      # Cleanup some discrepancies
+      options[:on] = options.delete('pit_date')
```

Comment:
Should this maybe be
```suggestion
      options[:on] = options.delete('pit_date') if options.key?('pit_date')
```
If `on` is defined, but `pit_date` isn't, I think this will erase the `on`.

Reply:
On will never be defined and pit_date better be.  This is just some compatibility stuff so the old (2018) PIT report will load.